examples/test_util/EclIO.cpp

```diff
@@ -0,0 +1,1179 @@
+}   
+
+std::vector<std::string> EclIO::split_string(const std::string &instr){
+
+    std::istringstream iss(instr);
+
+    std::vector<std::string> tokens{std::istream_iterator<std::string>{iss},
+        std::istream_iterator<std::string>{}};     
+ 
+    return tokens;
+}
+
+
+template <class T> std::string EclIO::make_scientific_string(T number){
+
+    std::stringstream buffer;
+    std::string str1,str2,exStr;
+    int p1,exp;
+    bool isDouble=false;
+    
+    if (sizeof(number)==8){
+        isDouble=true;
+    }
+    
+    if (isDouble==true){
+        buffer << std::uppercase << std::scientific << std::setprecision(13) << number << std::nouppercase;
+        exStr="D";
+    } else {
+        buffer << std::uppercase << std::scientific << std::setprecision(7) << number << std::nouppercase;
+        exStr="E";
+    }
+    
+    str1=buffer.str();
+    p1=str1.find_first_of("E");
+    
+    str1=str1.substr(0,p1);
+    p1=str1.find_first_of(".");
+    str1.erase(p1,1); 
+    str1.insert(p1-1,"0."); 
+
+    str2=buffer.str();
+
+    p1=str2.find_first_of("E");
+    str2=str2.substr(p1+1,str2.length()-p1-1);
+
+    exp=stoi(str2)+1;
+    
+    if (abs(exp)>99){
+       exStr="";
+    }
+    
+    buffer.str("");
+
+    if (abs(exp)<10){
+        if (exp < 0){
+	    buffer << str1 << exStr << "-0" << abs(exp); 
+        } else {
+	    buffer << str1 << exStr << "+0" << abs(exp); 
+        }
+    } else {
+        if (exp>0){
+	    buffer << str1 << exStr << "+" << exp; 
+        } else {
+	    buffer << str1 << exStr << exp; 
+	}    
+    }
+
+    return buffer.str();
+}
+
+
+void EclIO::readBinaryHeader(std::fstream &fileH,std::string &arrName,int &ant,std::string &arrType){
+
+    /* header data 4 byte (value 16), 8 byte (name of array), 4 byte number of elements, 4 byte array type 
+     ends with 4 byte (value should be 16 */
+
+    int bhead;
+    std::string tmpStrName(8,' ');
+    std::string tmpStrType(4,' ');
+
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+
+    fileH.read((char *)&bhead, sizeof(bhead));
+    bhead=reverseInt(bhead);
+    
+    if (bhead!=16){
+        std::string message="Error reading binary header. Expected 16 bytes of header data, found " + std::to_string(bhead);
+        throw std::runtime_error(message);
+    }
+    
+    fileH.read(&tmpStrName[0], 8);
+    
+    fileH.read((char *)&ant, sizeof(ant));
+    ant=reverseInt(ant);
+
+    fileH.read(&tmpStrType[0], 4);
+
+    fileH.read((char *)&bhead, sizeof(bhead));
+    bhead=reverseInt(bhead);
+    
+    if (bhead!=16){
+        std::string message="Error reading binary header. Expected 16 bytes of header data, found " + std::to_string(bhead);
+        throw std::runtime_error(message);
+    }
+    
+    arrName=tmpStrName;
+    arrType=tmpStrType;
+}
+
+
+void EclIO::stepOverArray(std::fstream &fileH,const int ant, const std::string arrType){
```

Comment:
This function is not generally usable.

Reply:
It is not clear to me what you mean by "not generally usable". I have renamed and modified the function.

  calcSizeOfArray(const int num, const EIOD::eclArrType &arrType)

Function is used to calculate number of bytes for storing an array of type eclArrType. This function will be used when scanning EclFiles. In stead of reading all elements, seekg(sizeOfArray) is used to set position of the file handler at the next header. 

See example  

    while (!file1.isEOF(&fileH)){
        file1.readBinaryHeader(fileH,arrName,num,arrType);
        sizeOfArray = file1.calcSizeOfArray(num,arrType); 

        if (sizeOfArray>0){      
            fileH.seekg(sizeOfArray,std::ios_base::cur);
        }
    }